cli/src/cmd/bind.rs

```diff
@@ -0,0 +1,184 @@
+        let c: Config = self.into();
+        c.out
+    }
+
+    /// Get the path to the root of the autogenerated crate
+    fn bindings_root(&self) -> PathBuf {
+        self.bindings_root.clone().unwrap_or_else(|| self.artifacts().join("bindings"))
+    }
+
+    /// Get the name of the autogenerated crate
+    fn crate_name(&self) -> String {
+        self.crate_name.clone().unwrap_or_else(|| "foundry-contracts".to_owned())
+    }
+
+    /// Get the version of the autogenerated crate
+    fn crate_version(&self) -> String {
+        self.crate_version.clone().unwrap_or_else(|| "0.0.1".to_owned())
+    }
+
+    /// `true` if the arguments set `crate_version` or `crate_name`
+    fn gen_crate(&self) -> bool {
+        self.crate_name.is_some() || self.crate_version.is_some()
+    }
+
+    /// `true` if the bindings root already exists
+    fn bindings_exist(&self) -> bool {
+        self.bindings_root().is_dir()
+    }
+
+    /// Instantiate the multi-abigen
+    fn get_multi(&self) -> eyre::Result<MultiAbigen> {
+        MultiAbigen::from_json_files(self.artifacts()).map_err(eyre::Report::msg)
+    }
+
+    /// Check that the existing bindings match the expected abigen output
+    fn check_existing_bindings(&self) -> eyre::Result<()> {
+        let bindings = self.get_multi()?.build().map_err(eyre::Report::msg)?;
+        println!("Checkign bindings for {} contracts", bindings.len());
+        if self.gen_crate() {
+            bindings
+                .ensure_consistent_crate(
+                    self.crate_name(),
+                    self.crate_version(),
+                    self.bindings_root(),
+                    self.single_file,
+                )
+                .map_err(eyre::Report::msg)?;
+        } else {
+            bindings
+                .ensure_consistent_module(self.bindings_root(), self.single_file)
+                .map_err(eyre::Report::msg)?;
+        }
+        Ok(())
+    }
+
+    /// Generate the bindings
+    fn generate_bindings(&self) -> eyre::Result<()> {
+        let bindings = self.get_multi()?.build().map_err(eyre::Report::msg)?;
+        println!("Generating bindings for {} contracts", bindings.len());
+        if self.gen_crate() {
+            bindings
+                .write_to_crate(
+                    self.crate_name(),
+                    self.crate_version(),
+                    self.bindings_root(),
+                    self.single_file,
+                )
+                .map_err(eyre::Report::msg)?;
+        } else {
+            bindings
+                .write_to_module(self.bindings_root(), self.single_file)
+                .map_err(eyre::Report::msg)?;
+        }
+        Ok(())
+    }
+}
+
+impl Cmd for BindArgs {
+    type Output = ();
+
+    fn run(self) -> eyre::Result<Self::Output> {
+        if !self.overwrite && self.bindings_exist() {
+            println!("Bindings found. Checking for consistency.");
+            return self.check_existing_bindings()
+        }
```

Comment:
nice, makes it very easy to run this on CI, eventually making a foundry + rust template repo using these would be sick

Reply:
I'm just exposing functionality someone else made :) it's a great feature to exist